Dockerfile.in

```diff
@@ -0,0 +1,212 @@
+# This file describes the standard way to build Docker, using docker
+#
+# Usage:
+#
+# # Assemble the full dev environment. This is slow the first time.
+# docker build -t docker .
+#
+# # Mount your source in an interactive container for quick testing:
+# docker run -v `pwd`:/go/src/github.com/docker/docker --privileged -i -t docker bash
+#
+# # Run the test suite:
+# docker run --privileged docker hack/make.sh test
+#
+# # Publish a release:
+# docker run --privileged \
+#  -e AWS_S3_BUCKET=baz \
+#  -e AWS_ACCESS_KEY=foo \
+#  -e AWS_SECRET_KEY=bar \
+#  -e GPG_PASSPHRASE=gloubiboulga \
+#  docker hack/release.sh
+#
+# Note: Apparmor used to mess with privileged mode, but this is no longer
+# the case. Therefore, you don't have to disable it anymore.
+#
+
+if [ "$ARCH" == "IBMPOWER" ]; then
+	FROM gccgo_image_14
```

Comment:
This strange mix of shell scripting and `Dockerfile` scripting is really, really hard to read IMO, and will be hard to maintain because I don't know of any editor that supports highlighting this sanely.  If we go with this, we might as well have just added `IF`/`ELSE` to `Dockerfile`s.